js/packages/common/src/actions/naming-service.ts

```diff
@@ -13,7 +17,53 @@ export const getTwitterHandle = async (
     );
     return twitterHandle;
   } catch (err) {
-    console.warn(`err: `, err);
     return undefined;
   }
 };
+
+export async function findOwnedNameAccountsForUser(
+  connection: Connection,
+  userAccount: PublicKey,
+): Promise<PublicKey[]> {
+  const NAME_PROGRAM_ID = new PublicKey(
+    'namesLPneVptA9Z5rqUDD9tMTWEJwofgaYwp8cawRkX',
+  );
+
+  const filters = [
+    {
+      memcmp: {
+        offset: 32,
+        bytes: userAccount.toBase58(),
+      },
+    },
+  ];
+  const accounts = await connection.getProgramAccounts(NAME_PROGRAM_ID, {
+    filters,
+  });
+  return accounts.map(a => a.pubkey);
+}
```

Comment:
Ah this is tough. We need to be really weary of using gpA its just too slow to use in anything that isn't backed by an index datastore.